xbmc/addons/interfaces/GUI/DialogFileBrowser.cpp

```diff
@@ -0,0 +1,341 @@
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  if (!strType || !additionalShare || !path_in || !path_out)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid handler data (additionalShare='%p', strType='%p', path_in='%p', path_out='%p') on addon '%s'",
+                          __FUNCTION__, additionalShare, strType, path_in, path_out, addon->ID().c_str());
+    return false;
+  }
+
+  std::string strPath = path_in;
+
+  VECSOURCES vecShares;
+  if (additionalShare)
+    GetVECShares(vecShares, additionalShare, strPath);
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetSource(strPath, allowNetworkShares, &vecShares, strType);
+  if (bRet)
+    *path_out = strdup(strPath.c_str());
+  return bRet;
+}
+
+bool Interface_GUIDialogFileBrowser::show_and_get_image(void* kodiBase, const char* shares, const char* heading,
+                                                        const char* path_in, char** path_out)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  if (!shares || !heading)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid handler data (shares='%p', heading='%p') on addon '%s'", __FUNCTION__, shares, heading, addon->ID().c_str());
+    return false;
+  }
+
+  std::string strPath = path_in;
+
+  VECSOURCES vecShares;
+  GetVECShares(vecShares, shares, strPath);
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetImage(vecShares, heading, strPath);
+  if (bRet)
+    *path_out = strdup(strPath.c_str());
+  return bRet;
+}
+
+bool Interface_GUIDialogFileBrowser::show_and_get_image_list(void* kodiBase, const char* shares, const char* heading, char*** path, unsigned int* entries)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  if (!shares || !heading)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid handler data (shares='%p', heading='%p') on addon '%s'", __FUNCTION__, shares, heading, addon->ID().c_str());
+    return false;
+  }
+
+  VECSOURCES vecShares;
+  GetVECShares(vecShares, shares, "");
+
+  std::vector<std::string> pathsInt;
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetImageList(vecShares, heading, pathsInt);
+  if (bRet)
+  {
+    *entries = pathsInt.size();
+    *path = static_cast<char**>(malloc(*entries*sizeof(char*)));
+    for (unsigned int i = 0; i < *entries; ++i)
+      (*path)[i] = strdup(pathsInt[i].c_str());
+  }
+  else
+    *entries = 0;
+  return bRet;
+}
+
+void Interface_GUIDialogFileBrowser::clear_file_list(void* kodiBase, char*** path, unsigned int entries)
```

Comment:
I would name the variable file_list, not path as it contains a list of several files, not a single path.